### PR TITLE
New potential

### PR DESCRIPTION
**Pull request by nathan-miller23**
* Updated .gitmodules to point to my new layouts branch in overcooked_env (will probably want to change this to point at master branch once overcooked_ai dev has stabilized)
* Better save and load utils
  * Now avoids loading in whole trainer and all worker envs, just loads policy directly
  * Simple, high level utils for loading an Overcooked Agent directly 
 ```
path = save(trainer)
agent = load_agent(path)
```

* Updated OvercookedMultiAgent env slightly to use new potential signature from overcooked_ai (potential function is now a function that must be called by client code instead of being parsed from info dict)

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -4,7 +4,7 @@
 [submodule "overcooked_ai"]
 	path = overcooked_ai
 	url = git@github.com:HumanCompatibleAI/overcooked_ai.git
-	branch = overcooked_ai_improvements
+	branch = new_mdp_layouts
```
Comment (micahcarroll) [inline .gitmodules:7]: Probably should be changed to master, assuming that PR is going to be merged soon

Comment (micahcarroll) [reply .gitmodules:7]: Ah, just say you already addressed this in the description – makes sense, i don't think it's urgent to change this

**Comment by micahcarroll** [comment]
We should probably get checks to pass actually before merging!

**Comment by mesutyang97** [comment]
@micahcarroll Looking at the checks, it seems like one of them is caused by `dill` and the other caused by `tensorflow`. How does one deal with those?

**Comment by micahcarroll** [comment]
I think Nathan should probably fix it. The dill issue can probably be simply fixed by adding it to the installation script.

File: .gitmodules (hunk 1 above)
Comment (nathan-miller23) [reply .gitmodules]: went ahead and changed it to master

**Comment by nathan-miller23** [comment]
Updated the install script to include dill, but still need to slightly modify the tests. Some of them are failing because of updates to Overcooked_ai that haven't been accounted for, and others are failing due to potential shaping changing the expected reward a sample trained agent gets. I say we merge overcooked_ai first (whenever it's ready) and then it should be quick to get this repo compatible with the newest and greatest overcooked_ai

**Comment by nathan-miller23** [comment]
All of the relevant updates to this PR were merged into [this](https://github.com/HumanCompatibleAI/human_aware_rl/pull/6) one

**Pull request closed without merging**